tests/components/homekit/test_homekit.py

```diff
@@ -526,3 +756,132 @@ def _mock_get_accessory(*args, **kwargs):
+
+    with patch(f"{PATH_HOMEKIT}.HomeKit") as mock_homekit:
+        mock_homekit.return_value = homekit = Mock()
+        type(homekit).async_start = AsyncMock()
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    mock_homekit.assert_any_call(
+        hass,
+        BRIDGE_NAME,
+        DEFAULT_PORT,
+        None,
+        ANY,
+        {},
+        DEFAULT_SAFE_MODE,
+        None,
+        None,
+        entry.entry_id,
+    )
+    assert mock_homekit().setup.called is True
+
+    # Test auto start enabled
+    mock_homekit.reset_mock()
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    await hass.async_block_till_done()
+
+    mock_homekit().async_start.assert_called()
+
+    migrated_persist_file_path = get_persist_fullpath_for_entry_id(hass, entry.entry_id)
+    assert (
+        await hass.async_add_executor_job(
+            json_util.load_json, migrated_persist_file_path
+        )
+        == legacy_homekit_state_contents
+    )
+    os.unlink(migrated_persist_file_path)
+    migrated_aid_file_path = get_aid_storage_fullpath_for_entry_id(hass, entry.entry_id)
+    assert (
+        await hass.async_add_executor_job(json_util.load_json, migrated_aid_file_path)
+        == legacy_homekit_aids_contents
+    )
+    os.unlink(migrated_aid_file_path)
+
+
+async def test_yaml_updates_update_config_entry_for_name(hass):
+    """Test async_setup with imported config."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        source=SOURCE_IMPORT,
+        data={CONF_NAME: BRIDGE_NAME, CONF_PORT: DEFAULT_PORT},
+        options={},
+    )
+    entry.add_to_hass(hass)
+
+    with patch(f"{PATH_HOMEKIT}.HomeKit") as mock_homekit:
+        mock_homekit.return_value = homekit = Mock()
+        type(homekit).async_start = AsyncMock()
+        assert await async_setup_component(
+            hass, "homekit", {"homekit": {CONF_NAME: BRIDGE_NAME, CONF_PORT: 12345}}
+        )
+        await hass.async_block_till_done()
+
+    mock_homekit.assert_any_call(
+        hass,
+        BRIDGE_NAME,
+        12345,
+        None,
+        ANY,
+        {},
+        DEFAULT_SAFE_MODE,
+        None,
+        None,
+        entry.entry_id,
+    )
+    assert mock_homekit().setup.called is True
+
+    # Test auto start enabled
+    mock_homekit.reset_mock()
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    await hass.async_block_till_done()
+
+    mock_homekit().async_start.assert_called()
+
+
+async def test_raise_config_entry_not_ready(hass):
+    """Test async_setup when the port is not available."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_NAME: BRIDGE_NAME, CONF_PORT: DEFAULT_PORT},
+        options={},
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.homekit.port_is_available", return_value=False,
+    ):
+        assert not await hass.config_entries.async_setup(entry.entry_id)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5453535)